R/functions_io.R

```diff
@@ -464,3 +464,138 @@ ParsePlateInformation = function(cell_names, pattern='^(\\S+)_(\\d+)_([A-Z])(\\d
+                               column_sample=column_sample,
+                               column_cluster=column_cluster,
+                               column_nUMI=paste("nCount",assay,sep="_"),
+                               column_nGene=paste("nFeature",assay,sep="_"),
+                               column_cell_cycle_seurat = column_ccphase,
+                               add_all_meta_data=T)
+  
+  # Export function does not include all information - need to add manually
+  crb_obj = readRDS(path)
+  
+  # Add basic parameters (are hardcoded unfortunately)
+  crb_obj$parameters[["gene_nomenclature"]] = "gene_name"
+  crb_obj$parameters[["discard_genes_expressed_in_fewer_cells_than"]] = "NA"
+  crb_obj$parameters[["keep_mitochondrial_genes"]] = TRUE
+  crb_obj$parameters[["variables_to_regress_out"]] = paste(param$vars_to_regress,collapse=",")
+  crb_obj$parameters[["number_PCs"]] = param$pc_n
+  crb_obj$parameters[["cluster_resolution"]] = param$cluster_resolution
+  crb_obj$parameters[["filtering"]] = list(UMI_min = "NA", UMI_max = "NA", genes_min = "NA", genes_max = "NA")
+  crb_obj$parameters[["tSNE_perplexity"]] = "NA"
+  
+  # Add gene lists (G2M_phase_genes, S_phase_genes, mitochondrial_genes may be hardcoded; additional lists may be custom)
+  if(!is.null(gene_lists) & length(gene_lists)>0) crb_obj$gene_lists = gene_lists
+  
+  # Add technical information
+  crb_obj$technical_info = list(R=R.utils::captureOutput(devtools::session_info()),seurat_version=packageVersion("Seurat"))
+  
+  # Add sample-related information
+  crb_obj$samples = list()
+  crb_obj$samples[["colors"]] = param$col_samples
+  crb_obj$samples[["overview"]] = data.frame(sample=levels(sc[[column_sample, drop=TRUE]]))
+  
+  crb_obj$samples[["by_cluster"]] = data.frame(sample = sc[[column_sample, drop=TRUE]], cluster=sc[[column_cluster, drop=TRUE]]) %>%
+    dplyr::group_by(sample,cluster) %>%
+    dplyr::summarise(num_cells=length(cluster)) %>%
+    dplyr::group_by(sample) %>%
+    dplyr::mutate(total_cell_count=sum(num_cells)) %>%
+    tidyr::pivot_wider(names_from=cluster, values_from=num_cells, values_fill=0)
+  
+  if ("Phase" %in% colnames(sc[[]])) {
+    crb_obj$samples[["by_cell_cycle_seurat"]] = data.frame(sample = sc[[column_sample, drop=TRUE]], phase=sc[[column_ccphase, drop=TRUE]]) %>% 
+      dplyr::group_by(sample,phase) %>%
+      dplyr::summarise(num_cells=length(phase)) %>%
+      dplyr::group_by(sample) %>%
+      dplyr::mutate(total_cell_count=sum(num_cells)) %>%
+      tidyr::pivot_wider(names_from=phase, values_from=num_cells, values_fill=0)
+  }
+  
+  # Add cluster-related information
+  crb_obj$clusters = list()
+  crb_obj$clusters[["colors"]] = param$col_clusters
+  crb_obj$clusters[["overview"]] = data.frame(cluster=levels(sc[[column_cluster, drop=TRUE]]))
+  
+  crb_obj$clusters[["by_samples"]] = data.frame(sample = sc[[column_sample, drop=TRUE]], cluster=sc[[column_cluster, drop=TRUE]]) %>%
```

Comment:
sc[["seurat_clusters"]] works? Is this the same as sc[[]]$seurat_clusters, or sc[[]][,"seurat_clusters"]?

Reply:
Yes :-)

Reply:
Ok, didn't know. I thought sc[["RNA"]] and so on can only be used for the assays.

Reply:
I do not know what happens if you add a metadata column with the name "RNA"....